### ISSUE TITLE
singularity run/shell/exec not respecting --shell option

### ISSUE THREAD

**Issue by bauerm97** [apptainer/singularity#2097]
## Version of Singularity:

```
mibauer@bauer-desktop:~$ singularity version
v3.0.0-beta.1-8-gccc859dd
```

### Expected behavior

```
mibauer@bauer-desktop:~$ singularity shell --shell /bin/bash test.sif 
Singularity test.sif:~> 
```

### Actual behavior

```
mibauer@bauer-desktop:~$ singularity shell --shell /bin/bash test.sif
Error: unknown flag: --shell
Usage:
  singularity [global options...] shell [shell options...] <container>
```

Right now, we're not respecting the `--shell` option on any of the actions [`run/exec/shell`]. There are two parts to this fix, the CLI side and the internal side.

1. CLI should be simple enough to fix, the flag already exists in `src/cmd/singularity/cli/action_flags.go`, it just needs to be added to the three action commands.
2. On the engine side, I think we just need to interject this in `StartProcess` @cclerget can advise.

**Comment by cclerget** [comment]
Fix planned with #2091

**Comment by cclerget** [comment]
Fixed by #2091

**Comment by ArangoGutierrez** [comment]
LOL
Thanks @cclerget
